src/cdk/scrolling/virtual-scroll-fixed-size.ts

```diff
@@ -0,0 +1,83 @@
+import {Range} from '@angular/cdk/collections';
+import {Directive, forwardRef, Input, OnChanges} from '@angular/core';
+import {VIRTUAL_SCROLL_STRATEGY, VirtualScrollStrategy} from './virtual-scroll-strategy';
+import {CdkVirtualScrollViewport} from './virtual-scroll-viewport';
+
+
+/** Virtual scrolling strategy for lists with items of known fixed size. */
+export class VirtualScrollFixedSizeStrategy implements VirtualScrollStrategy {
+  private _viewport: CdkVirtualScrollViewport;
+
+  constructor(public itemSize: number, public bufferSize: number) {}
+
+  /** Initialize the strategy and specify the viewport it will be working with. */
+  init(viewport: CdkVirtualScrollViewport) {
+    this._viewport = viewport;
+    this._viewport.totalContentSize = this._viewport.dataLength * this.itemSize;
+    this._updateRenderedRange();
+  }
+
+  /** Re-initialize the strategy with the same viewport. */
+  reinit() {
+    if (this._viewport) {
+      this.init(this._viewport);
+    }
+  }
+
+  onContentScrolled() {
+    this._updateRenderedRange();
+  }
+
+  onDataLengthChanged() {
+    this._viewport.totalContentSize = this._viewport.dataLength * this.itemSize;
+    this._updateRenderedRange();
+  }
+
+  private _updateRenderedRange() {
+    const scrollOffset = this._viewport.measureScrollOffset();
+    const firstVisibleIndex = Math.floor(scrollOffset / this.itemSize);
+    const range = this._expandRange(
+        {start: firstVisibleIndex, end: firstVisibleIndex},
+        this.bufferSize,
+        Math.ceil(this._viewport.viewportSize / this.itemSize) + this.bufferSize);
+    this._viewport.renderedRange = range;
+    this._viewport.renderedContentOffset = this.itemSize * range.start;
+  }
+
+  private _expandRange(range: Range, expandStart: number, expandEnd: number): Range {
+    const start = Math.max(0, range.start - expandStart);
+    const end = Math.min(this._viewport.dataLength, range.end + expandEnd);
+    return {start, end};
+  }
+}
+
+
+export function _virtualScrollFixedSizeStrategyFactory(fixedSizeDir: CdkVirtualScrollFixedSize) {
+  return fixedSizeDir._scrollStrategy;
+}
+
+
+/** A virtual scroll strategy that supports fixed-size items. */
+@Directive({
+  selector: 'cdk-virtual-scroll-viewport[itemSize]',
+  providers: [{
+    provide: VIRTUAL_SCROLL_STRATEGY,
+    useFactory: _virtualScrollFixedSizeStrategyFactory,
+    deps: [forwardRef(() => CdkVirtualScrollFixedSize)],
+  }],
+})
+export class CdkVirtualScrollFixedSize implements OnChanges {
+  /** The size of the items in the list. */
+  @Input() itemSize = 20;
```

Comment:
It would be nice if you could make this accept a function too where a user could pass a callback that accepted the current item they were iterating over to return a custom height for that element. For example:

```
itemSize = (item) => item.type === 'description' ? 50 : 20;
```

Reply:
Yep that's something I want to add in a future PR